Tests/SentryTests/Integrations/SentryUIPerformanceTrackerTests.swift

```diff
@@ -0,0 +1,234 @@
+
+        sut.viewControllerViewDidLoad(viewController) {
+            let blockSpan = self.getStack(tracker: tracker).last!
+            XCTAssertEqual(blockSpan.context.parentSpanId, transactionSpan.context.spanId)
+            XCTAssertEqual(blockSpan.context.spanDescription, self.fixture.viewDidLoad)
+        }
+        XCTAssertFalse(transactionSpan.isFinished)
+        
+        sut.viewControllerViewWillLayoutSubViews(viewController) {
+            let blockSpan = self.getStack(tracker: tracker).last!
+            XCTAssertEqual(blockSpan.context.parentSpanId, transactionSpan.context.spanId)
+            XCTAssertEqual(blockSpan.context.spanDescription, self.fixture.viewWillLayoutSubviews)
+        }
+        XCTAssertFalse(transactionSpan.isFinished)
+        
+        let layoutSubViewsSpan = self.getStack(tracker: tracker).last!
+        XCTAssertEqual(layoutSubViewsSpan.context.parentSpanId, transactionSpan.context.spanId)
+        XCTAssertEqual(layoutSubViewsSpan.context.spanDescription, self.fixture.layoutSubviews)
+        
+        sut.viewControllerViewDidLayoutSubViews(viewController) {
+            let blockSpan = self.getStack(tracker: tracker).last!
+            XCTAssertEqual(blockSpan.context.parentSpanId, transactionSpan.context.spanId)
+            XCTAssertEqual(blockSpan.context.spanDescription, self.fixture.viewDidLayoutSubviews)
+        }
+        XCTAssertFalse(transactionSpan.isFinished)
+        
+        sut.viewControllerViewWillAppear(viewController) {
+            let blockSpan = self.getStack(tracker: tracker).last!
+            XCTAssertEqual(blockSpan.context.parentSpanId, transactionSpan.context.spanId)
+            XCTAssertEqual(blockSpan.context.spanDescription, self.fixture.viewWillAppear)
+        }
+        XCTAssertFalse(transactionSpan.isFinished)
+        
+        sut.viewControllerViewDidAppear(viewController) {
+            let blockSpan = self.getStack(tracker: tracker).last!
+            XCTAssertEqual(blockSpan.context.parentSpanId, transactionSpan.context.spanId)
+            XCTAssertEqual(blockSpan.context.spanDescription, self.fixture.viewDidAppear)
+        }
+
+        XCTAssertEqual(Dynamic(transactionSpan).children.asArray!.count, 7)
+        XCTAssertTrue(transactionSpan.isFinished)
+    }
+    
+    func testTimeMeasurement() {
+        let sut = fixture.getSut()
+        let viewController = fixture.viewController
+        let tracker = fixture.tracker
+        var transactionSpan: Span!
+        
+        var date = Date(timeIntervalSince1970: 0)
+        
+        fixture.dateProvider.setDate(date: date)
+        var lastSpan: Span?
+        
+        sut.viewControllerLoadView(viewController) {
+            transactionSpan = self.getStack(tracker: tracker).first
+            lastSpan = self.getStack(tracker: tracker).last!
+            date = date.addingTimeInterval(1)
+            self.fixture.dateProvider.setDate(date: date)
```

Comment:
`l`: You could extract these two into a function and call it for example `advanceTimeBy`. I use this approach in many tests, see 
https://github.com/getsentry/sentry-cocoa/blob/ef7ba5fedd7afc40710567fdb7427375a3d52f29/Tests/SentryTests/Integrations/SentrySessionTrackerTests.swift#L331-L346